gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+    }
+    acceleratorTypes.sort(Comparator.comparing(AcceleratorType::getName));
+    return acceleratorTypes;
+  }
+
+  public List<Network> getNetworks(String projectId) throws IOException {
+    List<Network> networks = compute.networks().list(projectId).execute().getItems();
+
+    if (networks == null) {
+      return new ArrayList<>();
+    }
+    return networks;
+  }
+
+  public List<Subnetwork> getSubnetworks(String projectId, String networkSelfLink, String region)
+      throws IOException {
+    region = nameFromSelfLink(region);
+    List<Subnetwork> subnetworks =
+        compute.subnetworks().list(projectId, region).execute().getItems();
+    if (subnetworks == null) {
+      return new ArrayList<>();
+    }
+
+    // Only subnetworks in the parent network
+    Iterator it = subnetworks.iterator();
+    while (it.hasNext()) {
+      Subnetwork o = (Subnetwork) it.next();
+      if (!o.getNetwork().equals(networkSelfLink)) {
+        it.remove();
+      }
+    }
+
+    subnetworks.sort(Comparator.comparing(Subnetwork::getName));
+    return subnetworks;
+  }
+
+  public Operation insertInstance(String projectId, String template, Instance instance)
+      throws IOException {
+    final Compute.Instances.Insert insert =
+        compute.instances().insert(projectId, instance.getZone(), instance);
+    if (!Strings.isNullOrEmpty(template)) {
+      insert.setSourceInstanceTemplate(template);
+    }
+    return insert.execute();
+  }
+
+  public Operation terminateInstance(String projectId, String zone, String InstanceId)
+      throws IOException {
+    zone = nameFromSelfLink(zone);
+    return compute.instances().delete(projectId, zone, InstanceId).execute();
+  }
+
+  public Operation terminateInstanceWithStatus(
+      String projectId, String zone, String instanceId, String desiredStatus)
+      throws IOException, InterruptedException {
+    zone = nameFromSelfLink(zone);
+    Instance i = getInstance(projectId, zone, instanceId);
+    if (i.getStatus().equals(desiredStatus)) {
+      return compute.instances().delete(projectId, zone, instanceId).execute();
+    }
+    return null;
+  }
+
+  public Instance getInstance(String projectId, String zone, String instanceId) throws IOException {
+    zone = nameFromSelfLink(zone);
+    return compute.instances().get(projectId, zone, instanceId).execute();
+  }
+
+  /**
+   * Return all instances that contain the given labels
+   *
+   * @param projectId
+   * @param labels
+   * @return
+   * @throws IOException
+   */
+  public List<Instance> getInstancesWithLabel(String projectId, Map<String, String> labels)
+      throws IOException {
+    Compute.Instances.AggregatedList request = compute.instances().aggregatedList(projectId);
+    request.setFilter(buildLabelsFilterString(labels));
+    Map<String, InstancesScopedList> result = request.execute().getItems();
+    List<Instance> instances = new ArrayList<>();
+    for (InstancesScopedList instancesInZone : result.values()) {
```

Comment:
Why instancesInZone? Seems like matchingInstances would be more accurate.

Reply:
Done